support/main/default/classes/TriggerControlTest.cls

```diff
@@ -1,137 +1,243 @@
+    }
+
+    @isTest
+    private static void runAllEvents() {
+        Map<String, TriggerConfigService.ObjectConfig> mdt = new Map<String, TriggerConfigService.ObjectConfig>();
+        TriggerConfigService.ObjectConfig acct = generateObjTestData('account', true);
+        TriggerConfigService.HandlerConfig h3 = new TriggerConfigService.HandlerConfig();
+        h3.className = 'handler3';
+        h3.enabled = true;
+        h3.order = 0;
+        h3.autoBind = true;
+        h3.beforeInsert = true;
+        h3.afterInsert = true;
+        h3.beforeUpdate = true;
+        h3.afterUpdate = true;
+        h3.beforeDelete = true;
+        h3.afterDelete = true;
+        h3.afterUndelete = true;
+        acct = addChildHandlerToObj(acct, h3);
+        mdt.put('account', acct);
+        TriggerConfigService.configMetadata = mdt;
+        Id aId = createAccount();
+        Account a = new Account(Id = aId);
+
+        update a;
+        delete a;
+        undelete a;
+
+        // all events should have fired:
+        System.assert(Handler3.events.contains('beforeInsert'), 'beforeInsert should have fired');
+        System.assert(Handler3.events.contains('afterInsert'), 'afterInsert should have fired');
+        System.assert(Handler3.events.contains('beforeUpdate'), 'beforeUpdate should have fired');
+        System.assert(Handler3.events.contains('afterUpdate'), 'afterUpdate should have fired');
+        System.assert(Handler3.events.contains('beforeDelete'), 'beforeDelete should have fired');
+        System.assert(Handler3.events.contains('afterDelete'), 'afterDelete should have fired');
+        System.assert(Handler3.events.contains('afterUndelete'), 'afterUndelete should have fired');
+        System.assert(Handler3.events.size() == 7, 'All events should have fired one time');
+    }
+
+    // not sure how to implement a test on legacy binding methods without putting legacy binding in the trigger
+
+    // @isTest
+    // private static void legacyBindingTest() {
+    // }
+
+    @isTest
+    private static void invalidHandlerTest() {
+        Map<String, TriggerConfigService.ObjectConfig> mdt = buildMetadata();
+        TriggerConfigService.ObjectConfig objConfig = mdt.get('contact');
+        objConfig.disabledHandlers.clear();
+        objConfig.handlerOrder.clear();
+        objConfig.handlers.clear();
+        TriggerConfigService.HandlerConfig handlerConfig = generateHandlerTestData('badNameClass', true);
+        addChildHandlerToObj(objConfig, handlerConfig);
+        mdt.put('contact', objConfig);
+        TriggerConfigService.configMetadata = mdt;
+        // invalid class name should not run - currently just debugs
+        Boolean acctHandler1Fired = true;
+        Boolean acctHandler2Fired = true;
+        Boolean contactHandler1Fired = false;
+        runTest(acctHandler1Fired, acctHandler2Fired, contactHandler1Fired);
+    }
+
+    @isTest
+    private static void additionalCoverage() {
```

Comment:
needed? not sure coverage is an issue for us given this is packaged (and imo we should be making this a namespaced package)